examples/pytorch/ogb/ogbn-mag/hetero_rgcn.py

```diff
@@ -14,44 +16,20 @@ def extract_embed(node_embed, input_nodes):
     emb = {}
     for ntype, nid in input_nodes.items():
         nid = input_nodes[ntype]
-        if ntype in node_embed:
-            emb[ntype] = node_embed[ntype][nid]
+        if ntype in node_embed.embeds:
+            emb[ntype] = node_embed.embeds.pop(ntype)(nid)
     return emb
 
-class RelGraphEmbed(nn.Module):
-    r"""Embedding layer for featureless heterograph.
-    
-    Parameters
-    ----------
-    g : DGLGraph
-        Input graph.
-    embed_size : int
-        The length of each embedding vector
-    exclude : list[str]
-        The list of node-types to exclude (e.g., because they have natural features)
-    """
-    def __init__(self, g, embed_size, exclude=list()):
-        
-        super(RelGraphEmbed, self).__init__()
-        self.g = g
-        self.embed_size = embed_size
-
-        # create learnable embeddings for all nodes, except those with a node-type in the "exclude" list
-        self.embeds = nn.ParameterDict()
-        for ntype in g.ntypes:
-            if ntype in exclude:
-                continue
-            embed = nn.Parameter(th.Tensor(g.number_of_nodes(ntype), self.embed_size))
-            self.embeds[ntype] = embed
-
-        self.reset_parameters()
-
-    def reset_parameters(self):
-        for emb in self.embeds.values():
-            nn.init.xavier_uniform_(emb)
-
-    def forward(self, block=None):
-        return self.embeds
+def RelGraphEmbed(graph,embed_size,exclude):
+    node_num={}
+    for ntype in graph.ntypes:
+        if ntype in exclude:
+            continue
+        node_num[ntype] = graph.number_of_nodes(ntype)
+    embeds = HeteroEmbedding(node_num,embed_size)
+    for node_type in embeds.embeds.keys():
```

Comment:
You can directly add a `reset_parameters` function to the built-in `HeteroEmbedding` module.